### PR TITLE
Add lifecycle rostest

### PR DESCRIPTION
**Pull request by hidmic**
Connected to https://github.com/ros2/rostest/pull/1. This pull request adds a test equivalent to the lifecycle demo to exercise the `ros_testing` infrastructure.

### CODE REVIEW

File: lifecycle/test/test_lifecycle.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+# limitations under the License.
+
+import re
+import unittest
+
+import launch
+import launch.actions
+import launch.event_handlers.on_process_start
+
+
```
Comment (jacobperron) [inline lifecycle/test/test_lifecycle.py]: nit: remove extra empty line

File: lifecycle/test/test_lifecycle.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        package='lifecycle', node_executable='lifecycle_talker',
+        node_name='lc_talker', output='screen'
+    )
+    listener_node = launch_ros.actions.Node(
+        package='lifecycle', node_executable='lifecycle_listener',
+        node_name='listener', output='screen'
+    )
+    return launch.LaunchDescription([
+        talker_node, listener_node,
+        # When the talker reaches the 'inactive' state, make it take the 'configure' transition.
```
Comment (jacobperron) [inline lifecycle/test/test_lifecycle.py]: Is it really the "inactive" state? Referring to the [design doc](http://design.ros2.org/articles/node_lifecycle.html) I would expect the node to start in the "unconfigured" state.

File: lifecycle/test/test_lifecycle.py (hunk 1 above)
Comment (hidmic) [reply lifecycle/test/test_lifecycle.py]: Removed in https://github.com/ros2/demos/pull/336/commits/17757acdcc6c41779d87af75673d178de071e12d.

File: lifecycle/test/test_lifecycle.py (hunk 2 above)
Comment (hidmic) [reply lifecycle/test/test_lifecycle.py]: That's just a wrong comment. Revisited all of them in https://github.com/ros2/demos/pull/336/commits/17757acdcc6c41779d87af75673d178de071e12d. PTAL.

**Pull request merged**